aya-obj/src/btf/btf.rs

```diff
@@ -667,13 +675,58 @@ impl BtfExt {
         endianness: Endianness,
         btf: &Btf,
     ) -> Result<BtfExt, BtfError> {
-        // Safety: btf_ext_header is POD so read_unaligned is safe
+        #[repr(C)]
+        #[derive(Debug, Copy, Clone)]
+        struct MinimalHeader {
+            pub magic: u16,
+            pub version: u8,
+            pub flags: u8,
+            pub hdr_len: i32,
+        }
+
+        if data.len() < std::mem::size_of::<MinimalHeader>() {
+            return Err(BtfError::InvalidHeader);
+        }
+
+        // Safety: btf_ext_header and MinimalHeader are POD so read_unaligned is safe
         let header = unsafe {
-            ptr::read_unaligned::<btf_ext_header>(data.as_ptr() as *const btf_ext_header)
+            // first find the actual size of the header by converting into the minimal valid header
+            let minimal_header =
+                ptr::read_unaligned::<MinimalHeader>(data.as_ptr() as *const MinimalHeader);
+
+            // prevent invalid input from causing UB
+            if data.len() < minimal_header.hdr_len as usize {
```

Comment:
You also need to check that `minimal_header.hdr_len <= mem::size_of<btf_ext_header>()` right?

Reply:
Instead of checking I decided to use the minimum of the two so that the current header struct still works if newer programs have a bigger btf.ext header.